zipkin/src/main/java/zipkin2/internal/Proto3Fields.java

```diff
@@ -124,13 +124,18 @@ final void write(UnsafeBuffer b, T value) {
      * is returned when the length prefix is zero.
      */
     final T readLengthPrefixAndValue(UnsafeBuffer b) {
-      int length = readLengthPrefix(b);
+      int length = guardLength(b);
       if (length == 0) return null;
       return readValue(b, length);
     }
 
-    final int readLengthPrefix(UnsafeBuffer b) {
-      return b.readVarint32();
+    final int guardLength(UnsafeBuffer buffer) {
```

Comment:
unrelated.. I noticed that we weren't checking bounds in proto eventhough we were in thrift